aws/data_source_aws_dx_connection_test.go

```diff
@@ -0,0 +1,70 @@
+package aws
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAwsDxConnection_Basic(t *testing.T) {
+	key := "DX_LOCATION"
```

Comment:
Hmm. The acceptance tests for the corresponding hard-code the location to `EqSe2`, assuming that the AWS region is `us-west-2`. It's good to challenge that assumption.
Maybe call [`DescribeLocations`](https://docs.aws.amazon.com/directconnect/latest/APIReference/API_DescribeLocations.html) for the current test region and pick one at random rather than requiring an environment variable be set?

Reply:
Added https://github.com/terraform-providers/terraform-provider-aws/issues/9708 to track adding new data sources to make this easier.